### DISCUSSION TITLE
Standalone file support / detachedFiles

### DISCUSSION THREAD

**Discussion by jmn** [emacs-lsp/lsp-mode#3774] [Q&A]
I am wondering about the status of standalone file support in lsp-mode, which has been implemented in rust-analyzer.

For example when learning Rust by doing [rustlings](https://github.com/rust-lang/rustlings) it would be very convenient to have full language support for standalone files (unlinked-file) that are not in a project.

I am wondering exactly what needs to happen to get support for this implemented, and if there is anything I can do to make it happen.

- https://github.com/brotzeit/rustic/issues/391
- https://github.com/rust-lang/rust-analyzer/pull/8955

**Comment by yyoncho** [comment]
There is already a feature request for that. 

> I am wondering exactly what needs to happen to get support for this implemented, and if there is anything I can do to make it happen.

We have to support starting the language server with no root.

**Comment by flodiebold** [comment]
Note that you should be able to get rustlings working if you run [`rustlings lsp`](https://github.com/rust-lang/rustlings#enabling-rust-analyzer).
